source/faas-automated-testing/faas-automated-testing.rst

```diff
@@ -0,0 +1,299 @@
+| DRIVERS_ATLAS_PUBLIC_API_KEY  | The Atlas public API key            |
++-------------------------------+-------------------------------------+
+| DRIVERS_ATLAS_PRIVATE_API_KEY | The Atlas private API key           |
++-------------------------------+-------------------------------------+
+| DRIVERS_ATLAS_LAMBDA_USER     | The Atlas cluster user name         |
++-------------------------------+-------------------------------------+
+| DRIVERS_ATLAS_LAMBDA_PASSWORD | The Atlas cluster user password     |
++-------------------------------+-------------------------------------+
+| DRIVERS_ATLAS_GROUP_ID        | The driver's Atlas group id         |
++-------------------------------+-------------------------------------+
+| LAMBDA_STACK_NAME             | The driver's Lambda stack name      |
++-------------------------------+-------------------------------------+
+| AWS_REGION                    | The function AWS region             |
++-------------------------------+-------------------------------------+
+| AWS_ACCESS_KEY_ID             | Assume role atuomatically sets this |
++-------------------------------+-------------------------------------+
+| AWS_SECRET_ACCESS_KEY         | Assume role automatically sets this |
++-------------------------------+-------------------------------------+
+| AWS_SESSION_TOKEN             | Assume role automatically sets this |
++-------------------------------+-------------------------------------+
+
+
+This is an example function in the Evergreen config that accomplishes this, using
+subprocess.exec to execute a script that calls the drivers-evergreen-tools
+function inside of it:
+
+.. code:: yaml
+
+  run deployed aws lambda tests:
+  - command: ec2.assume_role
+    params:
+      role_arn: ${LAMBDA_AWS_ROLE_ARN}
+  - command: subprocess.exec
+    params:
+      working_dir: src
+      binary: bash
+      args:
+        - .evergreen/run-deployed-lambda-aws-tests.sh
+      env:
+        TEST_LAMBDA_DIRECTORY: ${PROJECT_DIRECTORY}/test/lambda
+        DRIVERS_TOOLS: ${DRIVERS_TOOLS}
+        DRIVERS_ATLAS_PUBLIC_API_KEY: ${DRIVERS_ATLAS_PUBLIC_API_KEY}
+        DRIVERS_ATLAS_PRIVATE_API_KEY: ${DRIVERS_ATLAS_PRIVATE_API_KEY}
+        DRIVERS_ATLAS_LAMBDA_USER: ${DRIVERS_ATLAS_LAMBDA_USER}
+        DRIVERS_ATLAS_LAMBDA_PASSWORD: ${DRIVERS_ATLAS_LAMBDA_PASSWORD}
+        DRIVERS_ATLAS_GROUP_ID: ${DRIVERS_ATLAS_GROUP_ID}
+        LAMBDA_STACK_NAME: dbx-node-lambda
+        AWS_REGION: us-east-1
+        AWS_ACCESS_KEY_ID: ${AWS_ACCESS_KEY_ID}
+        AWS_SECRET_ACCESS_KEY: ${AWS_SECRET_ACCESS_KEY}
+        AWS_SESSION_TOKEN: ${AWS_SESSION_TOKEN}
+
+The script itself:
+
+.. code:: none
+
+  #!/bin/bash
+  set -o errexit  # Exit the script with error if any of the commands fail
+  . ${DRIVERS_TOOLS}/.evergreen/run-deployed-lambda-aws-tests.sh
+
+Description of the behaviour of run-deployed-lambda-aws-tests.sh:
+
+- Creates a new Atlas cluster in a specific driver project
```

Comment:
Do we expect to run into any rate limits for cluster creation like we did during the astrolabe testing project?

Reply:
Added this section: https://github.com/mongodb/specifications/pull/1387/commits/f1ff81b7106843b6314c4f4959d24702732a352d